provisioning/README.md

```diff
@@ -0,0 +1,370 @@
+
+Binding Agents are needed by any Agent-Enabled dataspace providers to connect the dataspace protocol/representation (here: a combination of the SPARQL query language/operating on RDF triple graphs) to the underlying business data and logic.
+
+The Provisioning Agent in particular is able to interact with typical relational and structured backend sources based on SQL interfaces. The SPARQL profile which is used is called KA-BIND (the Knowledge Agent Binding Profile).
+
+Commercial alternatives to the FOSS Provisioning Agent are:
+* [GraphDB](https://www.ontotext.com/products/graphdb/)
+* [Neptune](https://aws.amazon.com/de/neptune/)
+* [Stardog](https://www.stardog.com/)
+* [Ontopic Studio](https://ontopic.ai/en/)
+
+## Architecture
+
+The FOSS Provisioning Agent uses the [OnTop Virtual Knowledge Graph](https://ontop-vkg.org/) system.
+
+![Provisioning Agent Architecture](../docs/ProvisioningOntop.drawio.svg)
+
+According to their homepage: "... exposes the content of arbitrary relational databases as knowledge graphs. These graphs are virtual, which means that data remains in the data sources instead of being moved to another database."
+
+Ontop operates on four standards: three W3C standards and one ANSI standard. It translates
+
+* incoming *Queries* in a subset/profile of [SparQL 1.1](https://www.w3.org/TR/sparql11-query/). We call that subset KA-BIND.
+* using a *Binding* (in Ontop language, its also called a *Mapping*) in [R2RML](https://www.w3.org/TR/r2rml/) (or the slightly simpler Ontop Mapping Language - OBDA)
+* from an *Ontology* in [OWL 2 QL](https://www.w3.org/TR/owl2-profiles/#OWL_2_QL) interpretation (and stored either in a Terse Triple Language - TTL or an RDF XML format)
+* into largely optimized [*SQL*](https://datacadamia.com/data/type/relation/sql/ansi) queries.
+
+The [Ontop CLI](https://ontop-vkg.org/tutorial/endpoint/endpoint-cli.html) is a Java/Spring application which must
+be extended with an appropriate JDBC driver and that can host only one endpoint per port. We have hence extended the original docker entrypoint scripting [resources](resources/entrypoint.sh) such that multiple endpoints and ports can be hosted in a single container. The following environment properties which were originally single-valued can now be set to arrays (where the last entry behaves as the original process to which the container liveness checks are tied to):
+* ONTOP_PORT - a list of port numbers to bind the services in the container to
+* ONTOP_ONTOLOGY_FILE - a list of local RDF resources (XML or TTL format) which describe the used (domain/use case) ontology. see the [Catena-X Ontology Repository](http://github.com/catenax-ng/product-ontology). For better performance, it is not recommended to include the complete Catena-X Ontology (ontology.ttl) but rather include one of the [Catena-X Use Case Ontology Folder](http://github.com/catenax-ng/product-ontology/tree/main/usecase)
```

Comment:
Do not link to catenax-ng repositories/GH Org. After consortia phase has ended this might be inaccessible.